File: decimal_binario.cpp
#include <stdio.h>

void convertirabinario(int n) {
    if (n == 0) {
        printf("0");
        return;
    }

    int binario[32];
    int indice = 0;

    while (n > 0)
        {
        binario[indice] = n % 2;
        n = n / 2;
        indice++;
        }


    for (int j = indice - 1; j >= 0; j--)
        {
        printf("%d", binario[j]);
    }

}

int main() {
    int numero = 8;
    printf("El binario de %d es: ", numero);
    convertirabinario(numero);

    return 0;
}

File: decimal_binario.py
def decimal_a_binario(n):
    if n == 0:
        return "0"

    binario = ""
    while n > 0:
        binario = str(n % 2) + binario
        n = n // 2

    return binario


numero = 8
print(f"El binario de {numero} es {decimal_a_binario(numero)}")

#corroboramos con la funcion nativa de python
print(f"comprobacion: El binario de {numero} es {bin(numero)[2:]}")